### PR TITLE
Fix generated metricbeat so create-metricset works.

### PR DESCRIPTION
**Pull request by blakerouse**
## What does this PR do?

Fixes the `make -C generator/_templates/metricbeat test` so that `make create-metricset` works inside of the generated custom metricbeat.

## Why is it important?

Both so `create-metricset` works on the generated beat and so the CI passes.

## Checklist

- [X] My code follows the style guidelines of this project
- [X] I have commented my code, particularly in hard-to-understand areas
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: metricbeat/scripts/mage/target/metricset/metricset.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+	devtools "github.com/elastic/beats/v7/dev-tools/mage"
+)
+
+// CreateMetricset creates a new metricset.
+//
+// Required ENV variables:
+// * MODULE: Name of the module
+// * METRICSET: Name of the metricset
+func CreateMetricset() error {
```
Comment (ycombinator) [inline metricbeat/scripts/mage/target/metricset/metricset.go:34]: For consistency with other targets' functions, how about taking a `context.Context` here as well?

Comment (blakerouse) [reply metricbeat/scripts/mage/target/metricset/metricset.go:34]: Being that it's not using it, I removed it. Should we have it there even if its not being used?

Comment (ycombinator) [reply metricbeat/scripts/mage/target/metricset/metricset.go:34]: Yeah, I could go either way over here. I see that other targets have it and it's not being used in them as well. Personally, I'd prefer to leave it out if it's not being used but perhaps we can do that everywhere in a follow up PR.

**Comment by ycombinator** [comment]
@blakerouse CI failure looks related.

**Comment by blakerouse** [comment]
Yeah its related, but do not know why its saying it is not present. From the commit you can see that it is present.

@kvch Any ideas why that would be the case?

**Comment by ycombinator** [comment]
@blakerouse I wonder if you need to split this PR up into two. First, one introducing the `github.com/elastic/beats/v7/metricbeat/scripts/mage/target/metricset` package. Once that's merged, then a second one that uses it.

File: metricbeat/scripts/mage/target/metricset/metricset.go [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	ve, err := devtools.PythonVirtualenv()
+	if err != nil {
+		return err
+	}
+	python, err := devtools.LookVirtualenvPath(ve, "python")
+	if err != nil {
+		return err
+	}
+	scriptPath := "scripts/create_metricset.py"
+	beatsBase := os.Getenv("ES_BEATS")
```
Comment (andrewkroh) [inline metricbeat/scripts/mage/target/metricset/metricset.go:44]: Don't rely on ES_BEATS. That's only by the Makefile's IIRC. When users invoke `mage` directly it won't be set.

Instead we have a `devtools.ElasticBeatsDir()` that can be used.

File: metricbeat/scripts/mage/target/metricset/metricset.go [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+	if beatsBase != "" {
+		scriptPath = filepath.Join(beatsBase, scriptPath)
+	}
+	path, err := os.Getwd()
```
Comment (andrewkroh) [inline metricbeat/scripts/mage/target/metricset/metricset.go:48]: There's `devtools.CWD()` that just panics on error.

File: metricbeat/magefile.go [hunk 4]
```diff
@@ -48,6 +47,8 @@ import (
 	"github.com/elastic/beats/v7/dev-tools/mage/target/unittest"
 	// mage:import
 	_ "github.com/elastic/beats/v7/dev-tools/mage/target/compose"
+	// mage:import
+	_ "github.com/elastic/beats/v7/metricbeat/scripts/mage/target/metricset"
```
Comment (andrewkroh) [inline metricbeat/magefile.go:51]: Should xpack/metricbeat/magefile.go get this too?

File: metricbeat/scripts/mage/target/metricset/metricset.go (hunk 2 above)
Comment (blakerouse) [reply metricbeat/scripts/mage/target/metricset/metricset.go]: Nice, updated to use that.

File: metricbeat/scripts/mage/target/metricset/metricset.go (hunk 3 above)
Comment (blakerouse) [reply metricbeat/scripts/mage/target/metricset/metricset.go]: Switched to using that.

File: metricbeat/magefile.go (hunk 4 above)
Comment (blakerouse) [reply metricbeat/magefile.go:51]: Yes, I have added it there.

**Pull request merged**